### ISSUE TITLE
FlagName none and any in constraints

### ISSUE THREAD

**Issue by phadej** [haskell/cabal#6760]

**Comment by phadej** [comment]
Doesn't seem to be related, but sometehing to check:

```
2020-05-06T14:40:09.7728087Z Unit Tests
2020-05-06T14:40:09.7728269Z   UnitTests.Distribution.Client.ProjectConfig
2020-05-06T14:40:09.7728432Z     ProjectConfig printing/parsing round trip
2020-05-06T14:40:09.7728665Z       all:                                                                   FAIL (12.03s)
2020-05-06T14:40:09.7728897Z         *** Failed! Falsified (after 95 tests and 124 shrinks):
2020-05-06T14:40:09.7765891Z         ProjectConfig {projectPackages = [], projectPackagesOptional = [], projectPackagesRepo = [], projectPackagesNamed = [], projectConfigBuildOnly = ProjectConfigBuildOnly {projectConfigVerbosity = NoFlag, projectConfigDryRun = NoFlag, projectConfigOnlyDeps = NoFlag, projectConfigSummaryFile = [], projectConfigLogFile = NoFlag, projectConfigBuildReports = NoFlag, projectConfigReportPlanningFailure = NoFlag, projectConfigSymlinkBinDir = NoFlag, projectConfigOneShot = NoFlag, projectConfigNumJobs = NoFlag, projectConfigKeepGoing = NoFlag, projectConfigOfflineMode = NoFlag, projectConfigKeepTempFiles = NoFlag, projectConfigHttpTransport = Flag "*S", projectConfigIgnoreExpiry = NoFlag, projectConfigCacheDir = Flag "*n", projectConfigLogsDir = Flag "}", projectConfigClientInstallFlags = ClientInstallFlags {cinstInstallLibs = Flag False, cinstIgnoreProject = NoFlag, cinstEnvironmentPath = Flag "Y;3s]", cinstOverwritePolicy = Flag NeverOverwrite, cinstInstallMethod = Flag InstallMethodCopy, cinstInstalldir = NoFlag}}, projectConfigShared = ProjectConfigShared {projectConfigDistDir = NoFlag, projectConfigConfigFile = NoFlag, projectConfigProjectFile = NoFlag, projectConfigHcFlavor = NoFlag, projectConfigHcPath = NoFlag, projectConfigHcPkg = NoFlag, projectConfigHaddockIndex = NoFlag, projectConfigRemoteRepos = [], projectConfigLocalRepos = [], projectConfigLocalNoIndexRepos = [], projectConfigIndexState = NoFlag, projectConfigStoreDir = NoFlag, projectConfigConstraints = [(UserConstraint (UserQualified (UserQualSetup (PackageName "FKih-RB")) (PackageName "D-KUfqw")) (PackagePropertyFlags (fromList [(FlagName "none",(1,False)),(FlagName "qr_",(1,True))])),ConstraintSourceProjectConfig "TODO")], projectConfigPreferences = [], projectConfigCabalVersion = NoFlag, projectConfigSolver = NoFlag, projectConfigAllowOlder = Nothing, projectConfigAllowNewer = Nothing, projectConfigWriteGhcEnvironmentFilesPolicy = NoFlag, projectConfigMaxBackjumps = NoFlag, projectConfigReorderGoals = NoFlag, projectConfigCountConflicts = NoFlag, projectConfigFineGrainedConflicts = NoFlag, projectConfigMinimizeConflictSet = NoFlag, projectConfigStrongFlags = NoFlag, projectConfigAllowBootLibInstalls = NoFlag, projectConfigOnlyConstrained = NoFlag, projectConfigPerComponent = NoFlag, projectConfigIndependentGoals = NoFlag, projectConfigProgPathExtra = []}, projectConfigProvenance = fromList [], projectConfigAllPackages = PackageConfig {packageConfigProgramPaths = MapLast {getMapLast = fromList []}, packageConfigProgramArgs = MapMappend {getMapMappend = fromList []}, packageConfigProgramPathExtra = [], packageConfigFlagAssignment = fromList [(FlagName "_",(1,False)),(FlagName "aks",(1,False)),(FlagName "bdya",(1,False)),(FlagName "c",(1,False)),(FlagName "ck",(1,True)),(FlagName "clmtp",(1,False)),(FlagName "crlw",(1,True)),(FlagName "cvlub",(1,True)),(FlagName "cvw",(1,True)),(FlagName "dk",(1,True)),(FlagName "dx",(1,False)),(FlagName "es",(1,False)),(FlagName "eur",(1,False)),(FlagName "ggm",(1,True)),(FlagName "ix",(1,True)),(FlagName "j",(2,False)),(FlagName "k",(1,True)),(FlagName "khj",(1,False)),(FlagName "kj",(1,True)),(FlagName "kl",(1,True)),(FlagName "kudgy",(1,True)),(FlagName "l",(3,True)),(FlagName "lc",(1,True)),(FlagName "ljuln",(1,True)),(FlagName "llgnc",(1,False)),(FlagName "m",(1,True)),(FlagName "mfv",(1,True)),(FlagName "n-byv",(1,False)),(FlagName "nkdxc",(1,True)),(FlagName "nq-z",(1,True)),(FlagName "nrw",(1,False)),(FlagName "ntr",(1,False)),(FlagName "o-_o_",(1,False)),(FlagName "q",(2,True)),(FlagName "qi-e",(1,True)),(FlagName "qzkvi",(1,False)),(FlagName "r_iqg",(1,False)),(FlagName "re",(1,False)),(FlagName "sn",(1,False)),(FlagName "t",(1,True)),(FlagName "tehg",(1,False)),(FlagName "tnwfi",(1,False)),(FlagName "u",(1,True)),(FlagName "ud",(1,False)),(FlagName "ulsm",(1,True)),(FlagName "vssrt",(1,True)),(FlagName "vy",(1,True)),(FlagName "w",(1,True)),(FlagName "wasky",(1,True)),(FlagName "xdf--",(1,False)),(FlagName "xe",(1,False)),(FlagName "xo",(1,True)),(FlagName "ymgq",(1,False)),(FlagName "zile",(1,False))], packageConfigVanillaLib = NoFlag, packageConfigSharedLib = NoFlag, packageConfigStaticLib = NoFlag, packageConfigDynExe = NoFlag, packageConfigFullyStaticExe = NoFlag, packageConfigProf = NoFlag, packageConfigProfLib = NoFlag, packageConfigProfExe = NoFlag, packageConfigProfDetail = NoFlag, packageConfigProfLibDetail = NoFlag, packageConfigConfigureArgs = [], packageConfigOptimization = NoFlag, packageConfigProgPrefix = NoFlag, packageConfigProgSuffix = NoFlag, packageConfigExtraLibDirs = [], packageConfigExtraFrameworkDirs = [], packageConfigExtraIncludeDirs = [], packageConfigGHCiLib = NoFlag, packageConfigSplitSections = NoFlag, packageConfigSplitObjs = NoFlag, packageConfigStripExes = NoFlag, packageConfigStripLibs = NoFlag, packageConfigTests = NoFlag, packageConfigBenchmarks = NoFlag, packageConfigCoverage = NoFlag, packageConfigRelocatable = NoFlag, packageConfigDebugInfo = NoFlag, packageConfigRunTests = NoFlag, packageConfigDocumentation = NoFlag, packageConfigHaddockHoogle = NoFlag, packageConfigHaddockHtml = NoFlag, packageConfigHaddockHtmlLocation = NoFlag, packageConfigHaddockForeignLibs = NoFlag, packageConfigHaddockExecutables = NoFlag, packageConfigHaddockTestSuites = NoFlag, packageConfigHaddockBenchmarks = NoFlag, packageConfigHaddockInternal = NoFlag, packageConfigHaddockCss = NoFlag, packageConfigHaddockLinkedSource = NoFlag, packageConfigHaddockQuickJump = NoFlag, packageConfigHaddockHscolourCss = NoFlag, packageConfigHaddockContents = NoFlag, packageConfigHaddockForHackage = NoFlag, packageConfigTestHumanLog = NoFlag, packageConfigTestMachineLog = NoFlag, packageConfigTestShowDetails = NoFlag, packageConfigTestKeepTix = NoFlag, packageConfigTestWrapper = NoFlag, packageConfigTestFailWhenNoTestSuites = NoFlag, packageConfigTestTestOptions = [], packageConfigBenchmarkOptions = []}, projectConfigLocalPackages = PackageConfig {packageConfigProgramPaths = MapLast {getMapLast = fromList []}, packageConfigProgramArgs = MapMappend {getMapMappend = fromList []}, packageConfigProgramPathExtra = [], packageConfigFlagAssignment = fromList [(FlagName "_bs",(1,False)),(FlagName "_zi",(1,False)),(FlagName "a",(1,False)),(FlagName "aryyn",(1,False)),(FlagName "bzr",(1,False)),(FlagName "c",(1,True)),(FlagName "cc",(1,False)),(FlagName "co",(1,False)),(FlagName "d-t",(1,False)),(FlagName "dth-g",(1,True)),(FlagName "e",(1,False)),(FlagName "ecdo",(1,True)),(FlagName "efwm",(1,True)),(FlagName "eiuy",(1,False)),(FlagName "f_bnm",(1,True)),(FlagName "fi",(1,True)),(FlagName "g",(1,False)),(FlagName "g_w",(1,False)),(FlagName "g_y",(1,False)),(FlagName "gcean",(1,True)),(FlagName "ggn-",(1,False)),(FlagName "gx_ld",(1,True)),(FlagName "hpdli",(1,False)),(FlagName "huy",(1,True)),(FlagName "j",(2,True)),(FlagName "jw",(1,False)),(FlagName "jzmqf",(1,False)),(FlagName "knj",(1,True)),(FlagName "kocs",(1,True)),(FlagName "kqtei",(1,False)),(FlagName "kw",(1,True)),(FlagName "l",(1,True)),(FlagName "l_rpo",(1,False)),(FlagName "ly-f-",(1,True)),(FlagName "mak",(1,False)),(FlagName "mocd",(1,True)),(FlagName "mwc",(1,True)),(FlagName "nbeq",(1,True)),(FlagName "ne",(1,False)),(FlagName "nekzr",(1,True)),(FlagName "ospf",(1,False)),(FlagName "p",(1,True)),(FlagName "pfu",(1,True)),(FlagName "pic",(1,True)),(FlagName "qe",(1,False)),(FlagName "r",(1,True)),(FlagName "rtck",(1,True)),(FlagName "sbho",(1,False)),(FlagName "t-k",(1,True)),(FlagName "tbtr",(1,False)),(FlagName "th",(1,False)),(FlagName "tt",(1,False)),(FlagName "u_",(1,False)),(FlagName "uc",(1,False)),(FlagName "uxh",(1,False)),(FlagName "uyen",(1,False)),(FlagName "vhmct",(1,True)),(FlagName "vlg",(1,False)),(FlagName "w",(1,False)),(FlagName "waj",(1,False)),(FlagName "we",(1,True)),(FlagName "xdzhp",(1,True)),(FlagName "zfx",(1,False))], packageConfigVanillaLib = NoFlag, packageConfigSharedLib = NoFlag, packageConfigStaticLib = NoFlag, packageConfigDynExe = NoFlag, packageConfigFullyStaticExe = NoFlag, packageConfigProf = NoFlag, packageConfigProfLib = NoFlag, packageConfigProfExe = NoFlag, packageConfigProfDetail = NoFlag, packageConfigProfLibDetail = NoFlag, packageConfigConfigureArgs = [], packageConfigOptimization = NoFlag, packageConfigProgPrefix = NoFlag, packageConfigProgSuffix = NoFlag, packageConfigExtraLibDirs = [], packageConfigExtraFrameworkDirs = [], packageConfigExtraIncludeDirs = [], packageConfigGHCiLib = NoFlag, packageConfigSplitSections = NoFlag, packageConfigSplitObjs = NoFlag, packageConfigStripExes = NoFlag, packageConfigStripLibs = NoFlag, packageConfigTests = NoFlag, packageConfigBenchmarks = NoFlag, packageConfigCoverage = NoFlag, packageConfigRelocatable = NoFlag, packageConfigDebugInfo = NoFlag, packageConfigRunTests = NoFlag, packageConfigDocumentation = NoFlag, packageConfigHaddockHoogle = NoFlag, packageConfigHaddockHtml = NoFlag, packageConfigHaddockHtmlLocation = NoFlag, packageConfigHaddockForeignLibs = NoFlag, packageConfigHaddockExecutables = NoFlag, packageConfigHaddockTestSuites = NoFlag, packageConfigHaddockBenchmarks = NoFlag, packageConfigHaddockInternal = NoFlag, packageConfigHaddockCss = NoFlag, packageConfigHaddockLinkedSource = NoFlag, packageConfigHaddockQuickJump = NoFlag, packageConfigHaddockHscolourCss = NoFlag, packageConfigHaddockContents = NoFlag, packageConfigHaddockForHackage = NoFlag, packageConfigTestHumanLog = NoFlag, packageConfigTestMachineLog = NoFlag, packageConfigTestShowDetails = NoFlag, packageConfigTestKeepTix = NoFlag, packageConfigTestWrapper = NoFlag, packageConfigTestFailWhenNoTestSuites = NoFlag, packageConfigTestTestOptions = [], packageConfigBenchmarkOptions = []}, projectConfigSpecificPackage = MapMappend {getMapMappend = fromList []}}
2020-05-06T14:40:09.7773242Z         NoParse "constraints" 4
2020-05-06T14:40:09.7776312Z         Use --quickcheck-replay=391543 to reproduce.
```

**Comment by phadej** [comment]
This is resolved in #6589
